app/assets/javascripts/map.js

```diff
@@ -58,5 +58,10 @@ $(function() {
         $("#ajax-response").html("My shizz ipsizzle pimpin' fizzle amet, shizzle my nizzle crocodizzle adipiscing elit. Nullam sapizzle velit, izzle volutpizzle, suscipit quizzle, gravida vel, dang. Pellentesque you son of a bizzle fo shizzle. Break yo neck, yall erizzle. Phat izzle dolor dapibizzle fo tempus fo shizzle. Maurizzle pellentesque nibh fo shizzle crunk. That's the shizzle izzle fo shizzle my nizzle. Pellentesque funky fresh crazy pot. In ass that's the shizzle platea dictumst. Donec sheezy. Fo shizzle mah nizzle fo rizzle, mah home g-dizzle bizzle urna, pretium the bizzle, ghetto ac, eleifend daahng dawg, nunc. Fo shizzle suscipit. Fo shizzle my nizzle mammasay mammasa mamma oo sa velit sizzle purus.");
       });
     });
+    if(address != "") {
```

Comment:
I think right now this is causing an error in the javascript console if the page is loaded without an address. This is because `address` on line 61 is not defined. It is being defined on line 47 but that is inside the `.submit` callback (i.e. it can't travel left) so it is `undefined`. All of your logic looks right, you just need to set `address` right before line 61.